packages/utils/src/gasPrice.ts

```diff
@@ -84,115 +30,103 @@ const getBumpAmount = (chainId: number) => {
-        return {
-            maxFeePerGas: maxFee,
-            maxPriorityFeePerGas: maxFee
-        }
-    }
-
-    return result
-}
-
-const estimateMaxPriorityFeePerGas = async (publicClient: PublicClient) => {
-    try {
-        const maxPriorityFeePerGasHex = await publicClient.request({
-            method: "eth_maxPriorityFeePerGas"
-        })
-        return hexToBigInt(maxPriorityFeePerGasHex)
-    } catch {
-        return null
-    }
-}
-
 const getFallBackMaxPriorityFeePerGas = async (
     publicClient: PublicClient,
     gasPrice: bigint
-) => {
+): Promise<bigint> => {
     const feeHistory = await publicClient.getFeeHistory({
         blockCount: 10,
         rewardPercentiles: [20],
         blockTag: "latest"
     })
 
-    if (feeHistory.reward === undefined) {
+    if (feeHistory.reward === undefined || feeHistory.reward === null) {
         return gasPrice
     }
 
-    const feeAverage =
-        feeHistory.reward.reduce((acc, cur) => cur[0] + acc, 0n) / 10n
-    return feeAverage < gasPrice ? feeAverage : gasPrice
+    const feeAverage = feeHistory.reward.reduce((acc, cur) => cur[0] + acc, 0n) / 10n
+    return minBigInt(feeAverage, gasPrice)
 }
 
-const getGasPriceFromRpc = (publicClient: PublicClient) => {
-    try {
-        return publicClient.getGasPrice()
-    } catch {
-        return null
+/// Formula taken from: https://eips.ethereum.org/EIPS/eip-1559
+const getNextBaseFee = async (publicClient: PublicClient) => {
+    const block = await publicClient.getBlock({
+        blockTag: "latest"
+    })
+    const currentBaseFeePerGas = block.baseFeePerGas || await publicClient.getGasPrice()
+    const currentGasUsed = block.gasUsed
+    const gasTarget = block.gasLimit / 2n
+
+    if (currentGasUsed === gasTarget) {
+        return currentBaseFeePerGas
+    }
+
+    if (currentGasUsed > gasTarget) {
+        const gasUsedDelta = currentGasUsed - gasTarget
+        const baseFeePerGasDelta = maxBigInt(currentBaseFeePerGas * gasUsedDelta / gasTarget / 8n, 1n)
+        return currentBaseFeePerGas + baseFeePerGasDelta
     }
+
+    const gasUsedDelta = currentGasUsed - gasTarget
+    const baseFeePerGasDelta = currentBaseFeePerGas * gasUsedDelta / gasTarget / 8n
+    return currentBaseFeePerGas - baseFeePerGasDelta
 }
 
 export async function getGasPrice(
-    chainId: number,
+    chain: Chain,
     publicClient: PublicClient,
+    noEip1559Support: boolean,
     logger: Logger
 ): Promise<GasPriceParameters> {
-    if (chainId === ChainId.Polygon || chainId === ChainId.Mumbai) {
-        const polygonEstimate = await getPolygonGasPriceParameters(
-            chainId,
-            logger
-        )
-        if (polygonEstimate) {
-            return polygonEstimate
+    let maxFeePerGas: bigint | undefined
+    let maxPriorityFeePerGas: bigint | undefined
+    if (noEip1559Support) {
+        let { gasPrice } = await publicClient.estimateFeesPerGas({ chain, type: "legacy" })
```

Comment:
should we not be try catching all these network calls? or do you think this is not required?